taqueria-vscode-extension-web-ui/src/data-editors/DataEditorNode.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { BoolEditor } from './BoolEditor';
+import { ListEditor } from './ListEditor';
+import { MapEditor } from './MapEditor';
+import { getFriendlyDataType } from './MichelineEditor';
+import { OptionEditor } from './OptionEditor';
+import { PairEditor } from './PairEditor';
+import { PrimitiveEditor } from './PrimitiveEditor';
+
+export const DataEditorNode = (
+	{ dataType, value, onChange, hideDataType }: {
+		dataType: any;
+		value: any;
+		onChange: (value: any) => void;
+		hideDataType?: boolean | undefined;
+	},
+) => (
+	<table>
+		<tr>
+			{hideDataType !== true && (
+				<td>
+					<h4 style={{ display: 'inline' }}>{getFriendlyDataType(dataType)}</h4>
+				</td>
+			)}
+			<td>
+				{getEditor({ dataType, onChange, value })}
+			</td>
+		</tr>
+	</table>
+);
+
+const getEditor = (
+	{ dataType, value, onChange }: { dataType: any; value: any; onChange: (value: any) => void },
+) => {
+	console.log(value);
+	const prim = dataType.prim;
+	switch (prim) {
+		case 'unit':
+			if (!value || typeof value !== 'object' || value.prim !== 'Unit') {
```

Comment:
I think this expression can be simplified to `typeof value !== 'object' || value.prim !== 'Unit'`?